### ISSUE TITLE
Add support for private image registries

### ISSUE THREAD

**Issue by AlexIoannides** [bodywork-ml/bodywork-core#203]
_"As a ML Engineer, I would like to be able to use Bodywork images hosted on a private image registry, so that I can use images that have passed security scanning as mandated by my place of work._"

**Tasks**
* Investigate how image pull secrets could be configured for private image repositories. Could we use the existing secrets mechanism?
* Investigate how images hosted on private image repositories can be pinged to check that they exist.
* Plan the work required to develop the required functionality.

**Comment by ansonchang** [comment]
YES , I also expect this function to try the amazing library. Because we are not allow to use public hub in work place. It will be great that we can try this function in private registry

**Comment by AlexIoannides** [comment]
Noted!
